File: LeetCode/Java/src/MergeTwoSortedList.java
import models.ListNode;

public class MergeTwoSortedList {
    public static ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        if (list1 != null && list2 != null) {
            if (list1.getVal() < list2.getVal()) {
                list1.setNext(mergeTwoLists(list1.getNext(), list2));
                return list1;
            } else
                list2.setNext(mergeTwoLists(list1, list2.getNext()));
            return list2;
        }
        if (list1 == null)
            return list2;
        else return list1;
    }

    static void afficher(ListNode l) {
        if (l != null) {
            System.out.println(l.getVal());
            afficher(l.getNext());
        }
    }

    public static void main(String[] args) {
        ListNode l1 = new ListNode(1, new ListNode(2, new ListNode(4)));
        ListNode l2 = new ListNode(1, new ListNode(3, new ListNode(4)));
        ListNode l3 = mergeTwoLists(l1,l2);
        afficher(l3);
    }
}

File: LeetCode/C++/MergeTwoSortedList.cpp
#include "ListNode.h"
#include "iostream"
using namespace std;

class Solution {
public:
    static ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
        if(list1== nullptr) return list2;
        if(list2== nullptr) return list1;
        if(list1->val>list2->val) swap(list1,list2);

        ListNode* head=list1;

        while(list1 && list2){
            ListNode* temp= nullptr;
            while(list1 && list1->val<=list2->val){
                temp= list1;
                list1=list1->next;
            }
            temp->next = list2;
            swap(list1,list2);
        }
        return head;
    }
};

/*
int main(){
    auto *list1=new ListNode(1);
    list1->next=new ListNode(2);
    list1->next->next=new ListNode(4);

    auto *list2=new ListNode(1);
    list2->next=new ListNode(3);
    list2->next->next=new ListNode(4);

    ListNode *list3=Solution::mergeTwoLists(list1,list2);
    while (list3!=nullptr){
        cout << list3->val;
        list3=list3->next;
    }
    return 0;
}*/
